### PR TITLE
NiMesh import and updates for newer xml.

### PR DESCRIPTION
**Pull request by Candoran2**
@niftools/blender-niftools-addon-reviewer 

# Overview
1. Updates to constraint import.
2. Made Nif file glob partially dependent on the xml-generated extensions.
3. Added support for imports of NiMeshes.
   1. Support for regular NiMesh import and Epic Mickey (2) NiMesh import.
   2. Support for weighted NiMeshes.
4. Updates to collision export.
5. Support for import of BSDynamicTriShape.
6. Update to Object properties ui.
7. Moved node type to use for the root node back to object.

##  Detailed Description
1. Constraint import needed to be updated due to xml changes - the most of the constraint information is now stored in an info struct in the constraint block, rather than the block itself. The update only changed some of the syntax to agree with the xml, I have no idea how functional constraint import actually is at the moment.
2. The file glob used by Blender's import/export screen for nifs is now dynamically constructed from the version objects in the nif versions module, which should store all non-nif/kf extensions used by those versions. In practice this isn't exactly the case so there is still some hardcoding.
3. -
   1. Importing the geometry is relatively simple and relies on specific names associated with the datastreams. Mixing of POSITION and POSITION_BP is not supported with the current code, but I have no indication that this actually occurs - or indeed that for any name other than TEXCOORD there are multiple.
   2. The armature import can be classed in two different cases: regular, or epic mickey. In the regular case, weight information is stored in the same manner as geometry information, and bone rest position information is stored in the NiSkinningModifier. In the epic mickey case there is no pose information, weight information and the bone rest positions are stored in the extra_em_data struct. I've somewhat crudely injected code into the existing armature handler to extract the correct information, but there might be better or more elegant ways of handling this.
4. Mostly changes to reflect the newer xml, as well as removing some setting of fields because they are unused or already set correctly because of their defaults.
5. BSDynamicTriShape identical to BSTriShape, except for one part: it stores its vertex information in a separate vertices field with Vector4, rather than as a Vector3 in the Vertex Data field. This causes means some special cases for BSDynamicTriShape in the BSTriShape paths.
6. Some properties were made conditionally accessible in the ui because they are only really used for root objects - i.e. those with no parents.
7. This already used to be the case. It was previously moved to scene, but this isn't consistent with how the rest of the object properties are handled (e.g. Prn or BSInvMarker).

## Fixes Known Issues
1. #533
2. #421 (only import, and can't check because the nif isn't available anymore)
3. #407

## Documentation
**[Overview of updates to documentation]**

## Testing
**[Overview of testing required to ensure functionality is correctly implemented]**

### Manual
**[Set of steps to manually verify updates are working correctly]**

### Automated
**[List of tests run, updated or added to avoid future regressions]**

## Additional Information
**[Anything else you deem relevant]**

### CODE REVIEW

File: io_scene_niftools/modules/nif_export/property/object/__init__.py [hunk 1]
```diff
@@ -239,11 +239,10 @@ def export_weapon_location(self, n_root, root_obj):
         game = bpy.context.scene.niftools_scene.game
         if game in ('OBLIVION', 'FALLOUT_3', 'SKYRIM'):
             loc = root_obj.niftools.prn_location
```
Comment (HENDRIX-ZT2) [inline io_scene_niftools/modules/nif_export/property/object/__init__.py:241]: PRN_DICT is no longer needed to get the string_data?

File: io_scene_niftools/modules/nif_import/armature/__init__.py [hunk 2]
```diff
@@ -281,21 +311,28 @@ def argmax(values):
...
         # now check the sign
-        actual_value = trans[max_coord_ind]
+        actual_value = translation[max_coord_ind]
         # handle sign accordingly so negative indices map to the negative identifiers in list
         if actual_value < 0:
             max_coord_ind += 3
-        axis_indices.append(max_coord_ind)
+        return max_coord_ind
+
+    def get_forward_axis(self, n_bone, axis_indices):
```
Comment (HENDRIX-ZT2) [inline io_scene_niftools/modules/nif_import/armature/__init__.py:326]: cool!

File: io_scene_niftools/modules/nif_import/object/__init__.py [hunk 3]
```diff
@@ -146,7 +149,7 @@ def import_object_flags(n_block, b_obj):
 
         if hasattr(n_block, "data") and isinstance(n_block.data.consistency_flags, NifClasses.ConsistencyType):
             b_obj.niftools.consistency_flags = n_block.data.consistency_flags.name
-        if n_block.is_skin():
+        if n_block.is_skin() and hasattr(n_block, "skin_instance"):
```
Comment (HENDRIX-ZT2) [inline io_scene_niftools/modules/nif_import/object/__init__.py:152]: Should this perhaps go into is_skin()?

**Review by HENDRIX-ZT2 — APPROVED** [review]
Looks solid!

File: io_scene_niftools/modules/nif_import/object/__init__.py (hunk 3 above)
Comment (Candoran2) [reply io_scene_niftools/modules/nif_import/object/__init__.py:152]: For BSTriShapes and NiTriShapes, it already is. However, NiMeshes can also be skins (at least how I've put it in new-pyffi), and they don't have a skin_instance.

**Pull request merged**